### DISCUSSION TITLE
Error: 'IHostBuilder' does not contain a definition for 'UseLogging' after upgrading to 17.6

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#12324] [Q&A]
I just upgraded my V.S. 2022 to the last 17.6.0 version and I generated a new solution using the Uno wizard.

I didn't selected neither mobile nor wasm because a I need just Windows and Skia heads.

I've added support for DI & Serilog, and I'm getting this error:

I've updated nuget packages to the last version too.

>'IHostBuilder' does not contain a definition for 'UseLogging' and no accessible extension method 'UseLogging' accepting a first argument of type 'IHostBuilder' could be found (are you missing a using directive or an assembly reference?)

The OnLaunched method:

```
        protected override void OnLaunched(LaunchActivatedEventArgs args) {
            var builder = this.CreateBuilder(args)

                .Configure(host => host
#if DEBUG
				// Switch to Development environment when running in DEBUG
				.UseEnvironment(Environments.Development)
#endif
                    .UseLogging(configure: (context, logBuilder) => {
                        // Configure log levels for different categories of logging
                        logBuilder.SetMinimumLevel(
                            context.HostingEnvironment.IsDevelopment() ?
                                LogLevel.Information :
                                LogLevel.Warning);
                    }, enableUnoLogging: true)
                    .UseSerilog(consoleLoggingEnabled: true, fileLoggingEnabled: true)
                    .UseConfiguration(configure: configBuilder =>
                        configBuilder
                            .EmbeddedSource<App>()
                            .Section<AppConfig>()
                    )
                    .ConfigureServices((context, services) => {
                        // TODO: Register your services
                        //services.AddSingleton<IMyService, MyService>();
                    })
                );
            _window = builder.Window;

            Host = builder.Build();

            // Do not repeat app initialization when the Window already has content,
            // just ensure that the window is active
            if (_window.Content is not Frame rootFrame) {
                // Create a Frame to act as the navigation context and navigate to the first page
                rootFrame = new Frame();

                // Place the frame in the current Window
                _window.Content = rootFrame;
            }

            if (rootFrame.Content == null) {
                // When the navigation stack isn't restored navigate to the first page,
                // configuring the new page by passing required information as a navigation
                // parameter
                rootFrame.Navigate(typeof(MainPage), args.Arguments);
            }
            // Ensure the current window is active
            _window.Activate();
        }
```

**Comment by mcNets** [comment]
I've installed the new Uno Extensions 4.8.12.15, generate a new solution and now it works.

As I asked on discord, can I get rid of the MacCatalyst folder in the .Mobile project or even the whole Mobile project?

**Reply by rkeithhill-keysight** [reply]
We only care about creating desktop apps for Windows, Linux, and macOS.  It's odd to me that the Uno project template puts macCatalyst in a project named `.Mobile`. I expected it to be either `.MacOS` or `.MacCatalyst`.

**Reply by rkeithhill-keysight** [reply]
Updating to 4.8.12.15 did not work for me. I had to add the `Uno.Extensions` namespace to `GlobalUsings.cs`. 🤦‍♂️

**Reply by jeromelaban** [reply]
> As I asked on discord, can I get rid of the MacCatalyst folder in the .Mobile project or even the whole Mobile project?

You can get remove any target framework or head project present in the generated app, in the same way that the Wizard does not include targets when not selected during the app generation.

>  It's odd to me that the Uno project template puts macCatalyst in a project named .Mobile. I expected it to be either .MacOS or .MacCatalyst.

About macCatalyst being in the mobile project, this is the least misleading way for including macCatalyst without having too many projects in the solution. In the future, when all projects can folded into one, we'll be able to remove that `.Mobile` suffix, but it's not possible yet.

I'll let @nickrandolph comment about the logging issue.

**Reply by mcNets** [reply]
Hi @jeromelaban 

>You can get remove any target framework or head project present in the generated app, in the same way that the Wizard does not include targets when not selected during the app generation.

I've generate a new project using the wizard following the next steps:

- Startup type: Blank and click Customize
- Framework: Net 7.0
- Platforms: All of them unchecked. (Android, IOS, WebAssembly)
- Presentation MVVM
- Projects: None
- Testing: None
- Features: None
- Extensions -> Navigation: Blank, Logging:Console
- Application name
- Theme: Fluent

And this is the summary:

![image](https://github.com/unoplatform/uno/assets/24267381/5ee5abc2-08c2-4ac6-9950-fbe74207b75e)

I have attached the generated solution too.
[TestTemplate.zip](https://github.com/unoplatform/uno/files/11544520/TestTemplate.zip)

Is there a way to avoid the McCatalyst head to be added?

**Reply by jeromelaban** [reply]
Interesting, thanks for the update. @dansiegel may be interesting in this as well.
